secrets/localsecrets/localsecrets.go

```diff
@@ -44,46 +43,38 @@ import (
 )
 
 func init() {
-	secrets.DefaultURLMux().RegisterKeeper(SchemeString, &URLOpener{})
-	secrets.DefaultURLMux().RegisterKeeper(SchemeBase64, &URLOpener{base64: true})
+	secrets.DefaultURLMux().RegisterKeeper(Scheme, &URLOpener{})
 }
 
-// SchemeString/SchemeBase64 are the URL schemes localsecrets registers its URLOpener under on secrets.DefaultMux.
+// Scheme is the URL scheme localsecrets registers its URLOpener under on
+// secrets.DefaultMux.
 // See the package documentation and/or URLOpener for details.
 const (
-	SchemeString = "stringkey"
-	SchemeBase64 = "base64key"
+	Scheme = "base64key"
 )
 
-// URLOpener opens localsecrets URLs like "stringkey://mykey" and "base64key://c2VjcmV0IGtleQ==".
+// URLOpener opens localsecrets URLs like "base64key://smGbjm71Nxd1Ig5FS0wj9SlbzAIrnolCz9bQQ6uAhl4=".
 //
-// For stringkey, the first 32 bytes of the URL host are used as the symmetric
-// key for encryption/decryption.
-//
-// For base64key, the URL host must be a base64 encoded string; the first 32
-// bytes of the decoded bytes are used as the symmetric key for
-// encryption/decryption.
+// The URL host must be base64 encoded, and must decode to exactly 32 bytes.
+// If the URL host is empty (e.g., "base64key://"), a new random key is generated.
 //
 // No query parameters are supported.
-type URLOpener struct {
-	base64 bool
-}
+type URLOpener struct{}
 
 // OpenKeeperURL opens Keeper URLs.
 func (o *URLOpener) OpenKeeperURL(ctx context.Context, u *url.URL) (*secrets.Keeper, error) {
 	for param := range u.Query() {
 		return nil, fmt.Errorf("open keeper %v: invalid query parameter %q", u, param)
 	}
-	if o.base64 {
-		sk, err := Base64Key(u.Host)
-		if err != nil {
-			return nil, fmt.Errorf("open keeper %v: base64 decode failed: %v", u, err)
-		}
-		return NewKeeper(sk), nil
+	var sk [32]byte
+	var err error
+	if u.Host == "" {
+		sk, err = NewRandomKey()
```

Comment:
is there any way for the user to get this randomly generated key back out again?

Reply:
No.